### PR TITLE
Take content out of selection area in RadioButton

### PR DESCRIPTION
**Pull request by DanielEliraz**
🔦 Summary
This is the description of RadioGroup selectionArea prop:

> Selection area emphasises the clickable area, none means no emphasis, hover is when the mouse is on the component, and always will show constantly

While the RadioButton content prop description is:

> node to be rendered under label. Clicking it will not trigger onChange

So I fixed that the content will be out of the selectionArea.

### CODE REVIEW

**Comment by EgoziE** [comment]
Under UX discussion

File: src/RadioGroup/RadioButton/RadioButton.driver.js [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 const radioButtonDriverFactory = ({ element }) => {
-  const radioButton = () => element.childNodes[0];
-  const label = () => element.childNodes[1];
+  const radioWrapper = () => element.childNodes[0];
+  const radioButton = () => radioWrapper().childNodes[0];
+  const label = () => radioWrapper().childNodes[1];
 
   return {
-    exists: () => !!element,
+    exists: () => !!radioWrapper(),
```
Comment (EgoziE) [inline src/RadioGroup/RadioButton/RadioButton.driver.js]: function exists is reserved for checking the element is not null (which in this case element.childNodes[0] will throw an error
Why do you need to check this child element specifically?

File: src/RadioGroup/RadioButton/RadioButton.driver.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
     check: () => ReactTestUtils.Simulate.change(radioButton()),
     isChecked: () => radioButton().checked,
     isDisabled: () => radioButton().disabled,
     getLabel: () => label().textContent,
-    getTabIndex: () => element.getAttribute('tabIndex'),
+    getLabelElement: () => label(),
+    getValue: () => radioButton().value,
```
Comment (EgoziE) [inline src/RadioGroup/RadioButton/RadioButton.driver.js]: Would you please add tests demonstrating usage of these new testkit functions?

File: src/RadioGroup/docs/examples.js [hunk 3]
```diff
@@ -19,3 +19,22 @@ export const selectionArea = `
     <RadioGroup.Radio value={2}>Option 2</RadioGroup.Radio>
   </RadioGroup>
 `;
+
+export const withContent = `
```
Comment (EgoziE) [inline src/RadioGroup/docs/examples.js]: Nice example, I think that the use case here will be demonstrated better with a selection area and node content:
```
  <RadioGroup value={1} selectionArea="always">
    <RadioGroup.Radio
      key={0}
      value={1}
      content={<Box margin={2}>Another details for option 1, not clickable</Box>}
    >
      Option 1
    </RadioGroup.Radio>
    <RadioGroup.Radio
      key={1}
      value={2}
      content={<Box margin={2}>Another details for option 2, not clickable</Box>}
    >
      Option 2
    </RadioGroup.Radio>
  </RadioGroup>
```

File: src/RadioGroup/docs/index.story.js [hunk 4]
```diff
@@ -45,6 +45,25 @@ const exampleChildren = [
       </RadioGroup.Radio>,
     ],
   },
+  {
+    label: 'Button with content',
+    value: [
+      <RadioGroup.Radio
+        key={0}
+        value={1}
+        content={'Another details for option 1, not clickable'}
```
Comment (EgoziE) [inline src/RadioGroup/docs/index.story.js]: Same comment as above:
<Box margin={2}>Another details for option 1, not clickable</Box>

File: src/RadioGroup/docs/index.story.js [hunk 5]
```diff
@@ -45,6 +45,25 @@ const exampleChildren = [
...
+      >
+        <div>Option 1</div>
+      </RadioGroup.Radio>,
+      <RadioGroup.Radio
+        key={1}
+        value={2}
+        content={'Another details for option 2, not clickable'}
```
Comment (EgoziE) [inline src/RadioGroup/docs/index.story.js]: Same comment as above:
<Box margin={2}>Another details for option 2, not clickable</Box>

File: src/RadioGroup/test/RadioGroup.visual.js [hunk 6]
```diff
@@ -88,17 +88,33 @@ const tests = [
...
+          },
+        },
+      },
+    ],
+  },
 ];
 
 tests.forEach(({ describe, its }) => {
-  its.forEach(({ it, props }) => {
+  its.forEach(({ it, props, radioButtonProps = {} }) => {
```
Comment (EgoziE) [inline src/RadioGroup/test/RadioGroup.visual.js]:
```suggestion
  its.forEach(({ it, props = {}, radioButtonProps = {} }) => {
```

File: src/RadioGroup/test/RadioGroup.visual.js [hunk 7]
```diff
@@ -88,17 +88,33 @@ const tests = [
       },
     ],
   },
+  {
+    describe: 'with content',
+    its: [
+      {
+        it: 'in 2 and 3 radios',
+        radioButtonProps: {
```
Comment (EgoziE) [inline src/RadioGroup/test/RadioGroup.visual.js]: Good visual test, I recommend adding a selection area:

```suggestion
        props: {
          selectionArea: 'always',
        },
        radioButtonProps: {
```

File: src/RadioGroup/RadioGroup.scss [hunk 8]
```diff
@@ -256,3 +256,7 @@ button.radioButton {
     background-color: $B60;
   }
 }
+
+.content {
+  margin-top: 6px;
```
Comment (EgoziE) [inline src/RadioGroup/RadioGroup.scss]: Is this a mandatory margin by design?
I think this should be style-less and users should decide for themselves their margin according the their design

Comment (DanielEliraz) [reply src/RadioGroup/RadioGroup.scss]: That's what Ayelet wanted

File: src/RadioGroup/RadioButton/RadioButton.driver.js (hunk 2 above)
Comment (DanielEliraz) [reply src/RadioGroup/RadioButton/RadioButton.driver.js]: It is used in the RadioGroup.driver.js

File: src/RadioGroup/RadioGroup.scss (hunk 8 above)
Comment (EgoziE) [reply src/RadioGroup/RadioGroup.scss]: cool

**Pull request merged**